Makefile

```
@@ -311,28 +309,24 @@ docker-test: ##@tests Run tests in a docker container with golang.
 
 test: test-unit ##@tests Run basic, short tests during development
 
-test-unit: SHELL := /bin/bash -o pipefail
-test-unit: UNIT_TEST_PACKAGES = $(shell go list ./...  | \
+test-unit: export UNIT_TEST_PACKAGES ?= $(shell go list ./... | \
 	grep -v /vendor | \
 	grep -v /t/e2e | \
 	grep -v /t/benchmarks | \
 	grep -v /transactions/fake )
 test-unit: ##@tests Run unit and integration tests
-	for package in $(UNIT_TEST_PACKAGES); do \
-		set -e; \
-		package_dir=$$(go list -f {{.Dir}} $${package}); \
-		go test -tags '$(BUILD_TAGS)' -timeout 30m -v -failfast $${package} $(gotest_extraflags) | \
-		  go-junit-report -iocopy -out $${package_dir}/report.xml; \
-	done
-
-test-unit-race: gotest_extraflags=-race
+	export BUILD_TAGS=$(BUILD_TAGS) ; \
+	export GOTEST_EXTRAFLAGS=$(GOTEST_EXTRAFLAGS) ; \
+	./_assets/scripts/run_unit_tests.sh
+
+test-unit-race: GOTEST_EXTRAFLAGS=-race
 test-unit-race: test-unit ##@tests Run unit and integration tests with -race flag
 
 test-e2e: ##@tests Run e2e tests
 	# order: reliability then alphabetical
 	# TODO(tiabc): make a single command out of them adding `-p 1` flag.
 
-test-e2e-race: gotest_extraflags=-race
+test-e2e-race: GOTEST_EXTRAFLAGS=-race
```

Same here, just use `export` here:
```suggestion
test-e2e-race: export GOTEST_EXTRAFLAGS = -race
```